src/app/wallets/get-on-chain-txs.ts

```diff
@@ -0,0 +1,24 @@
+import { BTC_NETWORK, ONCHAIN_MIN_CONFIRMATIONS, SECS_PER_10_MINS } from "@config"
+
+import { OnChainError, TxDecoder } from "@domain/bitcoin/onchain"
+import { CacheKeys } from "@domain/cache"
+
+import { RedisCacheService } from "@services/cache"
+import { OnChainService } from "@services/lnd/onchain-service"
+import { baseLogger } from "@services/logger"
+
+// we are getting both the transactions in the mempool and the transaction that
+// have been mined by not yet credited because they haven't reached enough confirmations
+export default async () =>
```

Comment:
this will be a big improvement and may help to not have any more errors when lnd1 is restarting